t/unit-tests/u-hashmap.c

```diff
@@ -165,39 +163,26 @@ static void t_add(struct hashmap *map, unsigned int ignore_case)
 
 		hashmap_for_each_entry_from(map, entry, ent)
 		{
-			int ret;
-			if (!check_int((ret = key_val_contains(
-						key_val, seen,
-						ARRAY_SIZE(key_val), entry)),
-				       ==, 0)) {
-				switch (ret) {
-				case 1:
-					test_msg("found entry was not given in the input\n"
-						 "    key: %s\n  value: %s",
-						 entry->key, get_value(entry));
-					break;
-				case 2:
-					test_msg("duplicate entry detected\n"
-						 "    key: %s\n  value: %s",
-						 entry->key, get_value(entry));
-					break;
-				}
+			int ret = key_val_contains(key_val, seen,
+							ARRAY_SIZE(key_val), entry);
+			cl_assert(ret == 0);
+			if (ret == 1) {
+				cl_check_fail("found entry was not given in the input\n");
+			} else if (ret == 2) {
+				cl_check_fail("duplicate entry detected\n");
 			} else {
 				count++;
 			}
 		}
-		check_int(count, ==, 2);
+		cl_assert_equal_i(count, 2);
 	}
 
 	for (size_t i = 0; i < ARRAY_SIZE(seen); i++) {
-		if (!check_int(seen[i], ==, 1))
-			test_msg("following key-val pair was not iterated over:\n"
-				 "    key: %s\n  value: %s",
-				 key_val[i][0], key_val[i][1]);
+		cl_assert_equal_i(seen[i], 1);
 	}
```

Comment:
You can also drop the braces around the for loop here -- we don't use them for single-line statements.